extensions/ql-vscode/src/remote-queries/download-link.ts

```diff
@@ -1,20 +1,25 @@
 /**
- * Represents a link to an artifact to be downloaded. 
+ * Represents a link to an artifact to be downloaded.
  */
 export interface DownloadLink {
   /**
-   *  A unique id of the artifact being downloaded. 
+   *  A unique id of the artifact being downloaded.
    */
   id: string;
 
   /**
    * The URL path to use against the GitHub API to download the
-   * linked artifact. 
+   * linked artifact.
    */
   urlPath: string;
 
   /**
    * An optional path to follow inside the downloaded archive containing the artifact.
    */
   innerFilePath?: string;
+
+  /**
+   * The full path to the directory where the artifacts for this link is stored.
+   */
+  artifactStorageDir: string;
```

Comment:
This doesn't feel like it belongs here to me. The download link is meant to hold information about where to find a download, not how to store it. 

How about we store it on the `RemoteQuery` entity itself? 

Or, as I mentioned in [this](https://github.com/github/vscode-codeql/pull/1142#discussion_r806610367) comment, perhaps we need something more central that manages where query data is stored rather than spreading the knowledge/responsibility around.

Reply:
I'll see what I can do. The difficulty is that there's no single place that is persistently accessible to make this information available in all places it's needed. It will need to be a bigger change. But, now that i have this piece in and it's working, I think I'm at a stage where I can add this abstraction.